src/test/java/io/conduit/SnowflakeDestinationIT.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2022 Meroxa, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.conduit;
+
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.time.Instant;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Sets;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import io.conduit.grpc.Data;
+import io.conduit.grpc.Destination;
+import io.conduit.grpc.Record;
+import io.grpc.stub.StreamObserver;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import static java.lang.System.currentTimeMillis;
+import static java.util.UUID.randomUUID;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class SnowflakeDestinationIT {
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private DestinationService underTest;
+
+    @BeforeEach
+    public void setUp() {
+        underTest = new DestinationService(new ClasspathTaskFactory());
+        cleanTable();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        cleanTable();
+    }
+
+    @SneakyThrows
+    private void cleanTable() {
+        try (var conn = getConnection();
+             var stmt = conn.prepareStatement("delete from CUSTOMERS_TEST")) {
+            stmt.execute();
+        }
+    }
+
+    // todo it's probably better to just fail the test, so it doesn't get silently ignored
```

Comment:
At least to me, these integration tests are closer to being a must-have than a nice-to-have.